com.ca.lsp.cobol/lsp-service-cobol/src/test/java/com/ca/lsp/cobol/service/DependencyServiceTest.java

```diff
@@ -44,13 +66,36 @@ public void initActivities() {
     dependencyService.setWorkspaceFolderPaths(Collections.singletonList(workspaceFolder));
   }
 
+  /**
+   * This test proves that the .dep file should update and remove the copybooks which were already
+   * downloaded and present in the .copybook directory
+   */
+  @Test
+  public void depFileUpdateTest() {
+    // create the .dep file
+    Path depFileReference = createDependencyFile(COPYBOOK_NAME);
+    dependencyService.addCopybookInDepFile(CPY_OUTER_NAME_ONLY2, DOCUMENT_URI);
+    int numberOfElements = getNumberOfElementsFromDepFile(depFileReference);
+
+    // trigger the event which will update the .dep file @link{CopybookDependencyServiceImpl.class}
+    broker.postData(
```

Comment:
It seems like we may call a `CopybookDependencyServiceImpl#obaserverCallback()` method directly, so we don't need a data bus broker and asynchronous awaiting.

Reply:
I agree if we have an "integration test" that use databus between ```DependencyService``` and ```CopybookService```

Reply:
> I made tests on che, here my result:
> 
> * the copybook is removed from dep file 5/5 times that I tried using zowe
> * the copybook is removed from dep file 5/5 times that I tried using a plain cobol file in the workspace
> 
> Two remarks:
> 
> * We still didn't clarify the scenario when the user change the name of a copybook previously defined (is still present in the dep file)
> * Using zowe some times the dependency folder/file is not created when open a cobol file. I had to refresh the browser and try again to have it working.

About the first remark, what I understood at our last meeting on Friday is not in priority for the first sprint and we should keep tracking it.
The second remark is it because of the issue that the IDE or CHE has it.
Thank you!

Reply:
In fact, the integration that should be asserted in this class is only a subscription to the data bus in the constructor, because we may rely on more specific data bus tests.

Reply:
👍